src/explorer.api/Explorers/EmailColumnExplorer.cs

```diff
@@ -0,0 +1,79 @@
+namespace Explorer
+{
+    using System;
+    using System.Collections.Generic;
+    using System.Linq;
+    using System.Threading;
+    using System.Threading.Tasks;
+
+    using Explorer.Diffix.Extensions;
+    using Explorer.Queries;
+
+    internal class EmailColumnExplorer : ExplorerBase
+    {
+        public const string EmailAddressChars = "abcdefghijklmnopqrstuvwxyzABCDEFGHIJKLMNOPQRSTUVWXYZ-_.";
+
+        public EmailColumnExplorer(IQueryResolver queryResolver, string tableName, string columnName)
+            : base(queryResolver)
+        {
+            TableName = tableName;
+            ColumnName = columnName;
+        }
+
+        private string TableName { get; }
+
+        private string ColumnName { get; }
+
+        public override async Task Explore(CancellationToken cancellationToken)
+        {
+            var emailCheckQ = await ResolveQuery<TextColumnTrim.Result>(
+                new TextColumnTrim(TableName, ColumnName, TextColumnTrimType.Both, EmailAddressChars),
+                cancellationToken);
+
+            var (totalValueCount, suppressedValueCount) = emailCheckQ.ResultRows.CountTotalAndSuppressed();
+
+            var isEmail = totalValueCount == emailCheckQ.ResultRows
+                .Where(r => r.TrimmedText == "@")
+                .Sum(r => r.Count);
```

Comment:
Maybe it's worth changing the Where ccondition to `Where(r => r.TrimmedText == "@" || r.IsNull)`. 
```suggestion
            var isEmail = totalValueCount == emailCheckQ.ResultRows
                .Where(r => r.TrimmedText == "@" || r.IsNull)
                .Sum(r => r.Count);
```

My thinking is that if the column is `NULL`, this has no impact on the types of values (ie. a column with a value of `NULL` might still be of type `email`). Also, the `totalValueCount` includes `NULL` values. 

Imagine this column, for example it's probably an email column:

| maybe_emails |
|----|
| `x@gmail.com` |
| `y@gmail.com` |
| `NULL` |
| `z@hotmail.com` |


Another option be to include a confidence value which could be the % of email-like columns.